frontend/src/views/EventDetail.vue

```diff
@@ -69,35 +69,54 @@ export default {
       if (!this.user) return 'You need to be logged in order to vote for this question.'
       return question.voted ? 'Unlike' : 'Like'
     },
-    handleVote(question) {
+    async handleVote(question) {
       if (!this.user) return
 
-      this.vote({ questionId: question._id, action: question.voted ? 'unlike' : 'like' })
+      try {
+        await this.vote({ questionId: question._id, action: question.voted ? 'unlike' : 'like' })
 
-      question.voted = !question.voted
-      question.votes += question.voted ? 1 : -1
-      this.sortQuestions()
+        question.voted = !question.voted
+        question.votes += question.voted ? 1 : -1
+        this.sortQuestions()
+      } catch (e) {
+        notification.error({ message: e.response?.data?.message ?? e.message ?? 'An unknown error occured' })
+      }
```

Comment:
Don't `await` vote,

Update the UI immediately, rollback if the request fails

Keep only necessary code inside try block